modules/gcp/compute.go

```diff
@@ -7,146 +7,259 @@ import (
+
+	ctx := context.Background()
+	service, err := NewComputeServiceE(t)
 	if err != nil {
 		t.Fatal(err)
 	}
-	return ip
+
+	// If we want to fetch an Instance without knowing its Zone, we have to query GCP for all Instances in the project
+	// and match on name.
+	instanceAggregatedList, err := service.Instances.AggregatedList(projectID).Context(ctx).Do()
+	if err != nil {
+		t.Fatalf("Instances.AggregatedList(%s) got error: %v", projectID, err)
+	}
+
+	for _, instanceList := range instanceAggregatedList.Items {
+		for _, instance := range instanceList.Instances {
+			if name == instance.Name {
+				return &Instance{projectID, instance}
+			}
+		}
+	}
+
+	t.Fatalf("Compute Instance %s could not be found in project %s", name, projectID)
+	return nil
+}
+
+// FetchImage queries GCP to return a new instance of the (GCP Compute) Image type
+func FetchImage(t *testing.T, projectID string, name string) *Image {
+	logger.Logf(t, "Getting Image %s", name)
+
+	ctx := context.Background()
+	service, err := NewComputeServiceE(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req := service.Images.Get(projectID, name)
+	image, err := req.Context(ctx).Do()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &Image{projectID, image}
 }
 
-// GetPublicIPOfInstanceE gets the public IP address of the given Instance in the given region.
-func GetPublicIPOfInstanceE(t *testing.T, projectID string, zone string, instanceID string) (string, error) {
-	logger.Logf(t, "Getting Public IP for Compute Instance %s", instanceID)
+// FetchRegionalInstanceGroup queries GCP to return a new instance of the Regional Instance Group type
+func FetchRegionalInstanceGroup(t *testing.T, projectID string, region string, name string) *RegionalInstanceGroup {
+	logger.Logf(t, "Getting Regional Instance Group %s", name)
 
 	ctx := context.Background()
+	service, err := NewComputeServiceE(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req := service.RegionInstanceGroups.Get(projectID, region, name)
+	instanceGroup, err := req.Context(ctx).Do()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return &RegionalInstanceGroup{projectID, instanceGroup}
+}
 
+// FetchZonalInstanceGroup queries GCP to return a new instance of the Regional Instance Group type
+func FetchZonalInstanceGroup(t *testing.T, projectID string, zone string, name string) *ZonalInstanceGroup {
+	logger.Logf(t, "Getting Zonal Instance Group %s", name)
+
+	ctx := context.Background()
 	service, err := NewComputeServiceE(t)
 	if err != nil {
-		return "", err
+		t.Fatal(err)
+	}
+
+	req := service.InstanceGroups.Get(projectID, zone, name)
+	instanceGroup, err := req.Context(ctx).Do()
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	instance, err := service.Instances.Get(projectID, zone, instanceID).Context(ctx).Do()
+	return &ZonalInstanceGroup{projectID, instanceGroup}
+}
+
+// GetPublicIP gets the public IP address of the given Compute Instance.
+func (c *Instance) GetPublicIp(t *testing.T) string {
```

Comment:
Please don't use short, non-descriptive names like `c`. 2 lines of code later and I have no idea what `c` is.

Reply:
Hm, in the all examples I read online (like [GoByExample](https://tour.golang.org/methods/3)), the idiomatic way to name the receiver variable is `i *Instance`. I even assume when I see a single-letter var when reading Go methods that it's the receiver var. 

That being said, I'll update the receiver vars to use the first letter of the object name (i.e. `i` instead of `c` for `*Instance`).